Lib/idlelib/tree.py

```diff
@@ -289,6 +289,11 @@ def drawtext(self):
             self.label.bind("<Button-4>", lambda e: wheel_event(e, self.canvas))
             self.label.bind("<Button-5>", lambda e: wheel_event(e, self.canvas))
         self.text_id = id
+        if self.dy == 0:
+            # The first row doesn't matter what the dy is, just measure its
+            # size to get the value of the subsequent dy
+            coords = self.canvas.bbox(id)
+            self.dy = coords[3] - coords[1]
```

Comment:
This results in too much space.  Adding ` - 3` works for me.  Also, since no complaints about 20 being too big, lets take max(20, <expression above with subtraction>)